sympy/assumptions/refine.py

```diff
@@ -70,6 +70,9 @@ def refine_abs(expr, assumptions):
         return arg
     if ask(Q.negative(arg), assumptions):
         return -arg
+    # arg is Mul
+    if isinstance(arg, Mul):
+        return refine(abs(arg.args[0]), assumptions) * refine(abs(arg / arg.args[0]), assumptions)
```

Comment:
I agree with your  suggestion.
Probably today I will not able to push the commit.